app/questionnaire/value_source_resolver.py

```diff
@@ -126,16 +126,18 @@ def _resolve_progress_value_source(
             return self.progress_store.get_section_status(identifier)
 
         if selector == "block":
+            if not self.location:
```

Comment:
You need to check rule evaluators set a location, `_is_section_enabled` currently doesn't, so this would error if you try to enable a section based on a block progress.

Reply:
`_is_section_enabled` now sets a location, as do the other RuleEvaluator initialisations